sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/action/QueryChangeStreamAction.java

```diff
@@ -62,6 +63,7 @@ public class QueryChangeStreamAction {
 
   private static final Logger LOG = LoggerFactory.getLogger(QueryChangeStreamAction.class);
   private static final Duration BUNDLE_FINALIZER_TIMEOUT = Duration.standardMinutes(5);
+  private static final Duration RESTRICTION_TRACKER_TIMEOUT = Duration.standardSeconds(40);
```

Comment:
Can you comment where this timeout comes from?  Ie what is the rpc deadline we are trying to avoid hitting that can currently cause an exception?  If the deadline is O(minutes) we might want to increase this to give the dofn the ability to catch up more if it is behind.

Reply:
Added comment. Basically we are trying to select a value lower than the predetermined deadline - https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java#L1859-L1871